editor/plugins/spatial_editor_plugin.cpp

```diff
@@ -5856,15 +5862,27 @@ void SpatialEditor::_init_grid() {
 
 void SpatialEditor::_finish_indicators() {
 	VisualServer::get_singleton()->free(origin_instance);
+	origin_instance = RID();
 	VisualServer::get_singleton()->free(origin);
+	origin = RID();
 
 	_finish_grid();
 }
 
 void SpatialEditor::_finish_grid() {
-	for (int i = 0; i < 3; i++) {
-		VisualServer::get_singleton()->free(grid_instance[i]);
-		VisualServer::get_singleton()->free(grid[i]);
+	for (int a = 0; a < 3; a++) {
+		if (grid_enable[a]) {
+			int c = (a + 2) % 3;
```

Comment:
This `if (grid_enable)` is new - just wondering why is this needed (sorry I'm not super familiar with this code)?